students/rhamersky/Lesson_3/mailroom_part1.py

```diff
@@ -0,0 +1,269 @@
+#!/usr/bin/env python3
+#  Description: This program will allow...........
+# Developer: Ryan Hamersky
+# Date: 05/10/2018
+# Rev: A - 05/15/2018 updated tup_donor_names so it would fit on one page and changed 'is' to '==' when comparing the
+#      variable intChoice.
+
+# -----Data Section-----
+strName = None     # --> User's name
+intOption = int()  # --> User's option.
+tup_donor_names = (["William Gates, III",1,10000.0],["Mark Zuckerberg",1,20000.0],
+                   ["Jeff Bezos",1,1200.0],["Paul Allen",1,5000.0],
+                   ["Mike Scott",1,25.0])  # --> Tuple that stores all donor's information
```

Comment:
not sure I really understand the purpose of this tuple yet.. you should use a list data structure as your global DB store, not a tuple which is immutable.

Reply:
Natasha,

I choose a tuple because it was immutable as I did not want to accidentally change the first element of the table when coding (list[0] = new_element). I only wanted to make the donor information mutable that is why I made the table a tuple and the row a list.

If this is not best practice in the programming field I can work on it, but being relative new to programming I did not want to make a costly mistake like this in this type of program.

Thanks,

Ryan

Reply:
`tuple` is a great data structure for something that needs to stay immutable. the donor data structure _is_ mutable as we are modifying it to add more donors.

it is pretty much expected for this to be a list (and a dictionary in next version of mailroom).